### PR TITLE
Windows toolchain fix

### LINKED ISSUE #29: pwasm-test doesn't run on windows
### Sample code

[step5 from tutorial](https://github.com/paritytech/pwasm-tutorial/tree/master/step-5)

### Expected result

Test passes

### Actual result:

```
libpwasm_ethereum-553f0e8a809c4035.rlib(pwasm_ethereum-553f0e8a809c4035.pwasm_ethereum5.rcgu.o): In function `pwasm_ethereum::ext::input::h8bbc80d741312350':
src/ext.rs:257: undefined reference to `input_length'
src/ext.rs:265: undefined reference to `fetch_input'
libpwasm_ethereum-553f0e8a809c4035.rlib(pwasm_ethereum-553f0e8a809c4035.pwasm_ethereum5.rcgu.o): In function `pwasm_ethereum::ext::ret::h7d6945f15c6ad3fb':
src/ext.rs:277: undefined reference to `ret'
```

Full log:

```
$ cargo test --features std
   Compiling pwasm-tutorial-contract v0.1.0 (file:///C:/Users/Alex/Documents/Repo/pwasm-tutorial/step-5)
error: linking with `gcc` failed: exit code: 1
  |
  = note: "gcc" "-Wl,--enable-long-section-names" "-fno-use-linker-plugin" "-Wl,--nxcompat" "-nostdlib" "-m64" "C:\\Users\\Alex\\.rustup\\toolchains\\nightly-2018-06-28-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\crt2.o" "C:\\Users\\Alex\\.rustup\\toolchains\\nightly-2018-06-28-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\rsbegin.o" "-L" "C:\\Users\\Alex\\.rustup\\toolchains\\nightly-2018-06-28-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\pwasm_tutorial_contract-496439e665775037.10ygozpgsu0ud2lu.rcgu.o" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\pwasm_tutorial_contract-496439e665775037.16u6js6g0l3k1ic6.rcgu.o" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\pwasm_tutorial_contract-496439e665775037.181cuta0v63atwcm.rcgu.o" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\pwasm_tutorial_contract-496439e665775037.1e4xbr21if77iprf.rcgu.o" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\pwasm_tutorial_contract-496439e665775037.1usgw96vgffjeez5.rcgu.o" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\pwasm_tutorial_contract-496439e665775037.1y16o1qfye96o7m0.rcgu.o" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\pwasm_tutorial_contract-496439e665775037.21ubqlz4vmrbhssm.rcgu.o" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\pwasm_tutorial_contract-496439e665775037.256dpilbml592mwe.rcgu.o" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\pwasm_tutorial_contract-496439e665775037.2eqka4xrczs0nsf1.rcgu.o" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\pwasm_tutorial_contract-496439e665775037.2fv2jzmxfiyps1xh.rcgu.o" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\pwasm_tutorial_contract-496439e665775037.2js0lmnwowiirzw3.rcgu.o" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\pwasm_tutorial_contract-496439e665775037.2lyh15q6cjwzy18c.rcgu.o" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\pwasm_tutorial_contract-496439e665775037.2xs38k3nu4oe9uo6.rcgu.o" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\pwasm_tutorial_contract-496439e665775037.3dph1ob44gcoxlug.rcgu.o" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\pwasm_tutorial_contract-496439e665775037.3jqq80ccf5axpodr.rcgu.o" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\pwasm_tutorial_contract-496439e665775037.3mjqqm2cnhilaoup.rcgu.o" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\pwasm_tutorial_contract-496439e665775037.3rngp6bm2u2q5z0y.rcgu.o" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\pwasm_tutorial_contract-496439e665775037.49a7n47po4ttqjl7.rcgu.o" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\pwasm_tutorial_contract-496439e665775037.49lx1q7cxvpykyv0.rcgu.o" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\pwasm_tutorial_contract-496439e665775037.4bsj7ns7sd673y62.rcgu.o" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\pwasm_tutorial_contract-496439e665775037.4cetbhrop16rnjw6.rcgu.o" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\pwasm_tutorial_contract-496439e665775037.4oc10dk278mpk1vy.rcgu.o" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\pwasm_tutorial_contract-496439e665775037.4xq48u46a1pwiqn7.rcgu.o" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\pwasm_tutorial_contract-496439e665775037.51s1w397y42gpez1.rcgu.o" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\pwasm_tutorial_contract-496439e665775037.6ltbk28ebvu4l8b.rcgu.o" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\pwasm_tutorial_contract-496439e665775037.8xzrsc1ux72v29j.rcgu.o" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\pwasm_tutorial_contract-496439e665775037.doar5ife0jn9b6h.rcgu.o" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\pwasm_tutorial_contract-496439e665775037.my259xyrfk9wtja.rcgu.o" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\pwasm_tutorial_contract-496439e665775037.y08g5q2x813c4wx.rcgu.o" "-o" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\pwasm_tutorial_contract-496439e665775037.exe" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\pwasm_tutorial_contract-496439e665775037.crate.allocator.rcgu.o" "-Wl,--gc-sections" "-nodefaultlibs" "-L" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps" "-L" "C:\\Users\\Alex\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\winapi-x86_64-pc-windows-gnu-0.4.0\\lib" "-L" "C:\\Users\\Alex\\.rustup\\toolchains\\nightly-2018-06-28-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "-Wl,-Bstatic" "C:\\Users\\Alex\\.rustup\\toolchains\\nightly-2018-06-28-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libtest-cdbcbae7d810ee45.rlib" "C:\\Users\\Alex\\.rustup\\toolchains\\nightly-2018-06-28-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libterm-d99af2ff1af42bc8.rlib" "C:\\Users\\Alex\\.rustup\\toolchains\\nightly-2018-06-28-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libgetopts-e03898247ed836ab.rlib" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\libpwasm_test-6191a873ff8dfdbd.rlib" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\libpwasm_ethereum-95f3f0ea9b1212f6.rlib" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\libpwasm_std-9a80e24453d19abc.rlib" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\libpwasm_abi-730e5ceff6bf4589.rlib" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\libpwasm_ethereum-553f0e8a809c4035.rlib" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\libpwasm_std-118210820fa0ee20.rlib" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\libtiny_keccak-af4a3b5c6a7cc496.rlib" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\libcfg_if-90068ea9e3729e9a.rlib" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\libparity_hash-f4256cdf57a7a462.rlib" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\libbigint-6cf49fd38e46a047.rlib" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\libcrunchy-05c1770358f54966.rlib" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\librustc_hex-6b5e10d9844955ed.rlib" "C:\\Users\\Alex\\Documents\\Repo\\pwasm-tutorial\\step-5\\target\\debug\\deps\\libbyteorder-419378a578212509.rlib" "-Wl,--start-group" "C:\\Users\\Alex\\.rustup\\toolchains\\nightly-2018-06-28-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libstd-1898c9c17939e020.rlib" "C:\\Users\\Alex\\.rustup\\toolchains\\nightly-2018-06-28-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libpanic_unwind-c1e3cbcc7fd502b7.rlib" "C:\\Users\\Alex\\.rustup\\toolchains\\nightly-2018-06-28-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libunwind-9481af84422c145d.rlib" "C:\\Users\\Alex\\.rustup\\toolchains\\nightly-2018-06-28-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\liblibc-b9992d68d7933ae6.rlib" "C:\\Users\\Alex\\.rustup\\toolchains\\nightly-2018-06-28-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\liballoc_system-b8ed53aff7766253.rlib" "C:\\Users\\Alex\\.rustup\\toolchains\\nightly-2018-06-28-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\liballoc-7b4e45dee4686adb.rlib" "C:\\Users\\Alex\\.rustup\\toolchains\\nightly-2018-06-28-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libcore-809f39cbcce011cd.rlib" "-Wl,--end-group" "C:\\Users\\Alex\\.rustup\\toolchains\\nightly-2018-06-28-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libcompiler_builtins-c6075c75264dac9b.rlib" "-Wl,-Bdynamic" "-l" "kernel32" "-l" "advapi32" "-l" "ws2_32" "-l" "userenv" "-l" "shell32" "-Wl,-Bstatic" "-l" "gcc_eh" "-l" "pthread" "-Wl,-Bdynamic" "-lmingwex" "-lmingw32" "-lgcc" "-lmsvcrt" "-lmsvcrt" "-luser32" "-lkernel32" "C:\\Users\\Alex\\.rustup\\toolchains\\nightly-2018-06-28-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\rsend.o"
  = note: C:\Users\Alex\Documents\Repo\pwasm-tutorial\step-5\target\debug\deps\libpwasm_ethereum-553f0e8a809c4035.rlib(pwasm_ethereum-553f0e8a809c4035.pwasm_ethereum5.rcgu.o): In function `pwasm_ethereum::ext::input::h8bbc80d741312350':
          C:\Users\Alex\.cargo\registry\src\github.com-1ecc6299db9ec823\pwasm-ethereum-0.5.1\src/ext.rs:257: undefined reference to `input_length'
          C:\Users\Alex\.cargo\registry\src\github.com-1ecc6299db9ec823\pwasm-ethereum-0.5.1\src/ext.rs:265: undefined reference to `fetch_input'
          C:\Users\Alex\Documents\Repo\pwasm-tutorial\step-5\target\debug\deps\libpwasm_ethereum-553f0e8a809c4035.rlib(pwasm_ethereum-553f0e8a809c4035.pwasm_ethereum5.rcgu.o): In function `pwasm_ethereum::ext::ret::h7d6945f15c6ad3fb':
          C:\Users\Alex\.cargo\registry\src\github.com-1ecc6299db9ec823\pwasm-ethereum-0.5.1\src/ext.rs:277: undefined reference to `ret'
```

### PR DESCRIPTION
**Pull request by Pzixel**
Fixes #29

### CODE REVIEW

File: src/externs.rs [hunk 1]
```diff
@@ -258,3 +258,24 @@ pub unsafe extern "C" fn elog(topic_ptr: *const u8, topic_count: u32, data_ptr:
 		r.borrow_mut().elog(topics, data);
 	});
 }
+
+#[cfg(windows)]
```
Comment (NikVolf) [inline src/externs.rs]: probably no need [cfg(windows)] here, would be useful for other platforms as well

Comment (NikVolf) [reply src/externs.rs]: linker on mac/linux probably just silently lets unresolved symbols to make their way to artefacts

Comment (Pzixel) [reply src/externs.rs]: As you with. I just didn't want to break something on your CI.

Comment (NikVolf) [reply src/externs.rs]: Thanks! Should be ok

**Comment by fckt** [comment]
perfect!

**Pull request merged**